### PR TITLE
Use new `ProgressOverlay`

### LINKED ISSUE #131: Add more useful messages + actions when an error occured
```
would something like 'find out more' or 'try again' be more useful to people? To explain what went wrong or prompt them to try again?
```

### LINKED ISSUE #141: Display Tx Hash as soon as possible
Use `isInBlock` instead of `isFinalized` as per comment here

Screen Shot 2022-03-31 at 09 22 04

https://github.com/cennznet/app-hub/pull/135#discussion_r838926078

### LINKED ISSUE #146: Display the correct value returned by extrinsic, as they include slippage
currently it shows the amounts that user puts in but does not accurately reflect the actual amounts have been taken

Screen Shot 2022-03-31 at 12 31 30 PM

### PR DESCRIPTION
**Pull request by ken-centrality**
## Description

Update the UI when transaction In Progress

#### Changes

- Use a new `ProgressOverlay` generic component for all sections
- Set to display "View Transaction" link ASAP
- Break `BridgeForm` into smaller parts

#### Fixes

- Closes #146 
- Closes #141
- Closes #131
 
Screen Shot 2022-04-05 at 16 06 34

## Notes

Because we want to grab the TX hash as soon as possible so there are substantial changes especially around the Bridge Deposit / Withdraw. Some good tests are needed

### CODE REVIEW

File: utils/CENNZTransaction.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+			const { event } = record;
+			return event?.section === eventSection && event?.method === eventMethod;
+		});
+
+		return record?.event;
+	}
+
+	getHashLink(): string {
+		return this.hash ? `${CENNZ_EXPLORER_URL}/extrinsic/${this.hash}` : null;
+	}
```
Comment (aidan-starke) [inline utils/CENNZTransaction.ts:70]: you can append `?network=Nikau` for Nikau explorer

**Review by aidan-starke — CHANGES_REQUESTED** [review]
lgtm! couple things,

 - [ ] dismiss button still missing for pool & swap
 - [ ] can we get the explorer link for CENNZ side of withdraw? would be helpful for users who need to do historical withdrawal in case of tx fail

File: utils/CENNZTransaction.ts (hunk 1 above)
Comment (ken-centrality) [reply utils/CENNZTransaction.ts:80]: yup, altho it doesn't work when you first going into it, needs to refresh once you are there 🤦

**Comment by ken-centrality** [comment]
@aidan-starke re having CENNZnet transaction link on the Bridge, I'm two minds about it, it's super useful for us to debug probably, but users would not really care, plus there are too many buttons on it already 😄.

Put a ticket to track that tho #157 157

**Comment by aidan-starke** [comment]
> @aidan-starke re having CENNZnet transaction link on the Bridge, I'm two minds about it, it's super useful for us to debug probably, but users would not really care, plus there are too many buttons on it already 😄.
> 
> Put a ticket to track that tho #157 157

no worries, it shouldn't be necessary with the new historic claim ui anyway

**Review by aidan-starke — APPROVED** [review]

File: utils/CENNZTransaction.ts (hunk 1 above)
Comment (aidan-starke) [reply utils/CENNZTransaction.ts:80]: oh weird

**Pull request merged**